diff -Naurw ./VT100/main.cpp ./VT100_usart/main.cpp
--- ./VT100/main.cpp	2021-02-13 15:24:25.000000000 +0100
+++ ./VT100_usart/main.cpp	2021-02-14 09:51:55.000000000 +0100
@@ -3,8 +3,7 @@
  * @brief Test VGA.
  * 
  */
-static usbd_device udev;
-static cdc_class   cdc_acm (udev);
+static UsartClass  usart (UsartNo2);
 static Console     console;
 //static Echo      echo;
 static GpioClass   LedGreen (GpioPortA, 10);
@@ -17,9 +16,8 @@
 }
 int main (void) {
   EnableDebugOnSleep();
-  udev.attach (cdc_acm);
-  cdc_acm.init();
-  console += cdc_acm;
+  usart.init (9600);
+  console += usart;
 //console += echo;
   +LedGreen;
   delay_frames (10);
diff -Naurw ./VT100/main.h ./VT100_usart/main.h
--- ./VT100/main.h	2021-02-10 18:16:51.000000000 +0100
+++ ./VT100_usart/main.h	2021-02-14 09:49:29.000000000 +0100
@@ -8,8 +8,7 @@
 #include "core_cm4.h"
 #include "system.h"
 #include "gpio.h"
-#include "usbd.h"
-#include "cdc_class.h"
+#include "usart.h"
 #include "console.h"
 /**
  * */
diff -Naurw ./VT100/Makefile ./VT100_usart/Makefile
--- ./VT100/Makefile	2021-02-13 13:22:08.000000000 +0100
+++ ./VT100_usart/Makefile	2021-02-14 09:50:47.000000000 +0100
@@ -7,9 +7,7 @@
 # S/C/C++ files
 SOURCES+= main.cpp
 SOURCES+= gpio.cpp
-SOURCES+= cdc_class.cpp
-SOURCES+= usbd_core.cpp
-SOURCES+= usbd_fs.cpp
+SOURCES+= usart.cpp
 SOURCES+= vhsync.cpp
 SOURCES+= vgaprint.cpp
 SOURCES+= keyboard.cpp
